python/ray/serve/_private/common.py

```diff
@@ -141,6 +199,251 @@ def update(
             message=message,
         )
 
+    def update_message(self, message: str):
+        return self._update(message=message)
+
+    def handle_transition(
+        self,
+        trigger: DeploymentStatusInternalTrigger,
+        message: str = "",
+        should_autoscale: bool = False,
+    ):
+        # If there was an unexpected internal error during reconciliation, set
+        # status to unhealthy immediately and return
+        if trigger == DeploymentStatusInternalTrigger.INTERNAL_ERROR:
+            return self._update(
+                status=DeploymentStatus.UNHEALTHY,
+                status_trigger=DeploymentStatusTrigger.INTERNAL_ERROR,
+                message=message,
+            )
+
+        # If deployment is being deleted, set status immediately and return
+        elif trigger == DeploymentStatusInternalTrigger.DELETE:
```

Comment:
Since all those are early returns, no need to chain the if-else statements together. Just a personal preference, would recommend to just use `if` here instead of `elif` so all the statements are independent. Same goes to all the ones below.